plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/templateFile.ts

```diff
@@ -0,0 +1,167 @@
+            type: 'string',
+          },
+          targetPath: {
+            title: 'Target Path',
+            description:
+              'Target path within the working directory to download the file as.',
+            type: 'string',
+          },
+          values: {
+            title: 'Template Values',
+            description: 'Values to pass on to the templating engine',
+            type: 'object',
+          },
+          cookiecutterCompat: {
+            title: 'Cookiecutter compatibility mode',
+            description:
+              'Enable features to maximise compatibility with templates built for fetch:cookiecutter',
+            type: 'boolean',
+          },
+          replace: {
+            title: 'Replace file',
+            description:
+              'If set, replace file in targetPath instead of overwriting existing one.',
+            type: 'boolean',
+          },
+          token: {
+            title: 'Token',
+            description:
+              'An optional token to use for authentication when reading the resources.',
+            type: 'string',
+          },
+        },
+      },
+    },
+    supportsDryRun: true,
+    async handler(ctx) {
+      ctx.logger.info('Fetching template file content from remote URL');
+
+      const workDir = await ctx.createTemporaryDirectory();
+      // Write to a tmp file, render the template, then copy to workspace.
+      const tmpFilePath = path.join(workDir, 'tmp');
+
+      const outputPath = resolveSafeChildPath(
+        ctx.workspacePath,
+        ctx.input.targetPath,
+      );
+
+      if (fs.existsSync(outputPath) && !ctx.input.replace) {
+        ctx.logger.info('File already exists in workspace, not replacing.');
+        return;
+      }
+
+      await fetchFile({
+        reader,
+        integrations,
+        baseUrl: ctx.templateInfo?.baseUrl,
+        fetchUrl: ctx.input.url,
+        outputPath: tmpFilePath,
+        token: ctx.input.token,
+      });
+
+      const { cookiecutterCompat, values } = ctx.input;
+      const context = {
+        [cookiecutterCompat ? 'cookiecutter' : 'values']: values,
+      };
+
+      ctx.logger.info(
+        `Processing template file with input values`,
+        ctx.input.values,
+      );
+
+      const renderTemplate = await SecureTemplater.loadRenderer({
+        cookiecutterCompat,
+        templateFilters: {
+          ...defaultTemplateFilters,
+          ...additionalTemplateFilters,
+        },
+        templateGlobals: additionalTemplateGlobals,
+        nunjucksConfigs: {
+          trimBlocks: ctx.input.trimBlocks,
+          lstripBlocks: ctx.input.lstripBlocks,
+        },
+      });
+
+      const contents = await fs.readFile(tmpFilePath, 'utf-8');
+      const result = renderTemplate(contents, context);
+      await fs.ensureDir(path.dirname(outputPath));
+      await fs.outputFile(outputPath, result);
+
+      ctx.logger.info(`Template result written to ${outputPath}`);
```

Comment:
What do you think to change the message more into the purpose of the action?

Something more like that the file has be created there.

Reply:
Done